File: Data Structures & Algorithms/non-cyclical-number/submission-5.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        count = 0

        while True:
            while n:
                dig = n % 10
                count += (dig)**2
                n //= 10
            
            if count == 1:
                return True
            elif count in seen:
                return False
            else:
                seen.add(count)
                n = count
                count = 0

File: Data Structures & Algorithms/two-integer-sum/submission-5.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> seen = new HashMap<>();

        for (int x = 0; x < nums.length; x++) {
            int diff = target - nums[x];

            if (seen.containsKey(diff)) {
                return new int[] {seen.get(diff), x};
            }

            seen.put(nums[x], x);
        }

        return new int[] {};
    }
}
